### DISCUSSION TITLE
commit signing of merge commits on pull

### DISCUSSION THREAD

**Discussion by crisdosaygo** [community/community#33119] [Other Feature Feedback, Questions, & Ideas] [answered]
Command line git will create an automatic merge commit for me when I pull down a remote branch that has diverged from what I have locally.

But after switching to commit signing I noticed that these merge commits are not signed (so I need to `git commit --amend --no-edit -S` to fix). 

Is there a git setting where I can instruct git to automatically sign these automatically created merge commits?

😨

**Comment by airtower-luna** [comment, marked as answer]
I would expect [`commit.gpgSign`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-commitgpgSign) to apply for those merges, too. If you want all commits to be signed it's convenient anyway.

Personally I prohibit pull-merges (`pull.ff=only`) and usually rebase no-yet-pushed changes, or merge manually where that's preferable.

**Reply by crisdosaygo** [reply]
That 

commit.gpgSign

is what I was looking for thank you! 🙂😜😉 Xx 😜
